packages/adaptive-ui-figma-designer/src/core/serialization.ts

```diff
@@ -1,112 +1,45 @@
-    designTokens: string;
-    componentAppliedStyleModules?: string;
-    appliedStyleModules: string;
-    componentAppliedDesignTokens?: string;
-    appliedDesignTokens: string;
-    additionalData: string;
-    effectiveAppliedStyleValues: string;
+export function mapReplacer(key: string, value: any): any {
+    if (value instanceof Map) {
+        return [["__dataType__", "Map"], ...value];
+        // Legacy format:
+        // return {
+        //     dataType: "Map",
+        //     value: [...value],
+        // };
+    } else {
+        return value;
+    }
 }
 
-/**
- * Serializable version of the Figma selected node state.
- */
-export interface SerializableUIState {
-    selectedNodes: SerializableNodeData[];
+export function mapReviver(key: string, value: any): any {
+    if (value !== null) {
+        if (typeof value === "object" && value.dataType === "Map") {
+            // Legacy format (revive existing values, see above)
+            return new Map(value.value);
+        } else if (Array.isArray(value) && value.length > 0 && value.every(val => Array.isArray(val) && val.length === 2)) {
+            if (value[0][0] === "__dataType__") {
+                value = value.slice(1);
+            }
+            return new Map(value);
+        }
+    }
+    return value;
 }
 
-/**
- * Converts node data from the UI to serializable format.
- * @param nodes Node data in the UI format.
- * @returns Node data in the serializable format.
- */
-export function serializeUINodes(
-    nodes: PluginUINodeData[]
-): SerializableNodeData[] {
-    const serializedNodes = nodes.map(
-        (node): SerializableNodeData => {
-            return {
-                id: node.id,
-                type: node.type,
-                name: node.name,
-                supports: node.supports,
-                children: serializeUINodes(node.children),
-                inheritedDesignTokens: (node.inheritedDesignTokens as DesignTokenValues).serialize(),
-                componentDesignTokens: (node.componentDesignTokens as DesignTokenValues)?.serialize(),
-                designTokens: node.designTokens.serialize(),
-                componentAppliedStyleModules: (node.componentAppliedStyleModules as AppliedStyleModules)?.serialize(),
-                appliedStyleModules: node.appliedStyleModules.serialize(),
-                componentAppliedDesignTokens: (node.componentAppliedDesignTokens as AppliedDesignTokens)?.serialize(),
-                appliedDesignTokens: node.appliedDesignTokens.serialize(),
-                additionalData: node.additionalData.serialize(),
-                effectiveAppliedStyleValues: node.effectiveAppliedStyleValues.serialize(),
-            };
+export function deserializeMap<K, V>(json?: string): Map<K, V> {
+    if (json) {
+        try {
+            const map = JSON.parse(json, mapReviver) as Map<K, V>;
+            return map;
+        } catch (e) {
+            // console.warn(e);
+            // Ignore, empty string
```

Comment:
Same as the other comment, do these comments need to be cleaned up?

Reply:
I made the check a bit more useful.